be/src/vec/functions/function_hash.h

```diff
@@ -205,4 +205,97 @@ class FunctionAnyHash : public IFunction {
         return Status::OK();
     }
 };
+
+template <typename Impl>
+class FunctionMurMurHash3 : public IFunction {
+public:
+    static constexpr auto name = Impl::name;
+    static FunctionPtr create() { return std::make_shared<FunctionMurMurHash3>(); }
+
+private:
+    using ToType = typename Impl::ReturnType;
+
+    template <bool first>
+    Status execute_murmur_hash3(const IColumn* column, typename ColumnVector<ToType>::Container& vec_to) {
+        if (const ColumnString* col_from = check_and_get_column<ColumnString>(column)) {
+            const typename ColumnString::Chars& data = col_from->get_chars();
+            const typename ColumnString::Offsets& offsets = col_from->get_offsets();
+            size_t size = offsets.size();
+
+            ColumnString::Offset current_offset = 0;
+            for (size_t i = 0; i < size; ++i) {
+                if (first) {
+                    const ToType seed = Impl::apply(reinterpret_cast<const char*>(&data[current_offset]),
+                                                    offsets[i] - current_offset - 1);
+                    vec_to[i] = seed;
+                } else
+                    vec_to[i] = Impl::combineHashes(ext::bit_cast<UInt32>(vec_to[i]),
+                                                    reinterpret_cast<const char*>(&data[current_offset]),
+                                                    offsets[i] - current_offset - 1);
+                current_offset = offsets[i];
+            }
+        } else if (const ColumnConst* col_from_const =
+                           check_and_get_column_const_string_or_fixedstring(column)) {
+            String value = col_from_const->get_value<String>().data();
+            if (first) {
+                const ToType seed = Impl::apply(value.data(), value.size());
+                const size_t size = vec_to.size();
+                vec_to.assign(size, seed);
+            } else {
+                for (size_t i = 0; i < vec_to.size(); ++i) {
+                    vec_to[i] = Impl::combineHashes(ext::bit_cast<UInt32>(vec_to[i]), value.data(), value.size());
+                }
+            }
+        } else {
+            LOG(FATAL) << fmt::format("Illegal column {} of first argument of function ",
+                                      column->get_name(), get_name());
+        }
+        return Status::OK();
+    }
+
+    Status execute_for_argument(const IColumn* column,
+                                typename ColumnVector<ToType>::Container& vec_to, bool& is_first) {
+        if (is_first)
+            execute_murmur_hash3<true>(column, vec_to);
+        else
+            execute_murmur_hash3<false>(column, vec_to);
+        is_first = false;
+        return Status::OK();
+    }
+
+public:
+    String get_name() const override { return name; }
+
+    bool is_variadic() const override { return true; }
+    size_t get_number_of_arguments() const override { return 0; }
```

Comment:
这一行是有问题的，参数个数不是0，是1或2吧？

Reply:
`For non-variadic functions, return number of arguments; otherwise return zero (that should be ignored)`

Reply:
`For non-variadic functions, return number of arguments; otherwise return zero (that should be ignored)`

> 看了一下函数的demo用法：输入是一个字符串或者两个字符串，输出是数字。
> 
> ```
> mysql> select murmur_hash3_32("hello");
> +--------------------------+
> | murmur_hash3_32('hello') |
> +--------------------------+
> |               1321743225 |
> +--------------------------+
> 
> mysql> select murmur_hash3_32("hello", "world");
> +-----------------------------------+
> | murmur_hash3_32('hello', 'world') |
> +-----------------------------------+
> |                         984713481 |
> +-----------------------------------+
> ```
> 
> 1.没必要单独写个类，输入一个字符串的可以继承FunctionUnaryToType，输入两个字符串的可以继承FunctionBinaryToType，然后用函数重载的方式，这些能力目前都是支持了的。不然实现风格不统一，后面的同学会觉得代码很混乱。 2.单侧你也单独写了一个be/test/vec/function/function_hash_test.cpp，可以一起放function_string_test.cpp里面。 3.另外单侧太少了，就两行。 4.疑问：当输入两个参数时，支持null吗？看你单侧里没写。
> 
> 1这一点要是嫌麻烦我可以帮你改 234你可以自己改

1. 我理解这个函数的输入参数的数量是不确定的，所以没法完全复用FunctionUnaryToType和FunctionBinaryToType的逻辑
2. hash function写单独的ut我感觉是make sense的，我重新搞了个文件是因为之前的murmur hash 2没写ut...
3. 单测我再补充一些，多列里包含一列或多列null的hash值都应该是null，我补一下

Reply:
ok